test/e2e/stop_test.go

```diff
@@ -107,6 +107,32 @@ var _ = Describe("Podman stop", func() {
 		Expect(output).To(ContainSubstring(cid3))
 	})
 
+	It("podman stop container --time", func() {
+		session := podmanTest.RunTopContainer("test4")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		cid1 := session.OutputToString()
+
+		session = podmanTest.Podman([]string{"stop", "--time", "1", "test4"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		output := session.OutputToString()
+		Expect(output).To(ContainSubstring(cid1))
+	})
+
+	It("podman stop container --timeout", func() {
+		session := podmanTest.RunTopContainer("test5")
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		cid1 := session.OutputToString()
+
+		session = podmanTest.Podman([]string{"stop", "--timeout", "1", "test5"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		output := session.OutputToString()
+		Expect(output).To(ContainSubstring(cid1))
+	})
+
```

Comment:
I'm not sure any of these tests would've caught the problem: that is, the problem is that the container is killed immediately, without a chance to trap SIGTERM. Could you try running these against a broken build? If I read them correctly, they will all pass.